### ISSUE TITLE
Multiple update statements with one round trip to database

### ISSUE THREAD

**Issue by MaklaCof** [zzzprojects/EntityFramework-Plus#211]
This code:
```
 foreach (var item in model)
                await this.context.Carts.Where(t => t.Id == item.id).UpdateAsync(t => new DbCart() { Order = item.order });

```
is much slower then this:
```
 var items = await this.context.Carts.Where(t => model.Select(t2 => t2.id).Contains(t.Id)).ToDictionaryAsync(t => t.Id, t => t);
            foreach (var item in model)
                items[item.id].Order = item.order;
            await this.context.SaveChangesAsync();

```
Is it possible to use update with only one round trip to database? Maybe future update or something like that.

**Comment by JonathanMagnan** [comment]
Hello @MaklaCof ,

Yes, it's normal that `SaveChanges` is faster due to how Batch Update SQL is executed. You normally use it when you have several items to update at once.

I have added this request in the roadmap. That will certainly be done when we will start to re-write our library for the v2.x version.

Best Regards,

Jonathan

**Comment by MaklaCof** [comment]
Thanks. Feel free to close the issue.

**Comment by granthoff1107** [comment]
@MaklaCof how is that even possible, since when does EF support multiple async request from the same context is this new?

**Comment by MaklaCof** [comment]
@granthoff1107 
I was not talking about multiple request (at the same time - parallel), but a possible way to prepare data and then post all changes with single request which is possible. EF does that when for example you add several new entities to DbSet and then call SaveChanges. Same applies for [BulkOperations](http://entityframework-extensions.net/).

**Comment by granthoff1107** [comment]
@MaklaCof  Yes, I'm aware of the original issue.  I'm just confused because you stated the original query is much slower than the later,  which implies that the original query was able to run in the first place.

I'm confused on how you got EntityFramework to run parallel Queries from the Same context.

**Comment by MaklaCof** [comment]
First query runs one query for every update (inside for), that's why it is much slower then second which runs once (on save changes). No parallel execution. As far as I know, this can no be done on same context. A new context must be created. [Link1](https://stackoverflow.com/questions/45742584/get-context-instance-from-dbcontextpool-ef-core-2-0-to-use-it-in-task), [Link2](https://stackoverflow.com/questions/41749896/ef-6-how-to-correctly-perform-parallel-queries).
